alonzo/impl/src/Cardano/Ledger/Alonzo/PlutusScriptApi.hs

```diff
@@ -153,7 +153,15 @@ collectTwoPhaseScriptInputs pp tx utxo =
     Just cost -> merge (apply cost) (map redeemer needed) (map getscript needed) (Right [])
   where
     txinfo = transTx utxo tx
-    needed = scriptsNeeded utxo tx
+    needed = filter knownToNotBe1Phase $ scriptsNeeded utxo tx
```

Comment:
What does this have to do with testing?

Reply:
this is a bug fix (that I have posted in https://github.com/input-output-hk/cardano-ledger-specs/pull/2272 as well). Tim was trying to generate valid Alonzo Txs, and I told him that that might help.